packages/kmod-5.10-nvidia/nvidia-tesla-build-config.toml.in

```diff
@@ -0,0 +1,18 @@
+[nvidia-tesla]
+lib-modules-path = "kernel/drivers/extra/video/nvidia/tesla"
+objects-source = "NVIDIA_MODULES"
+
+[nvidia-tesla.kernel-modules.nvidia]
+dependencies = ["nvidia.mod.o"]
+
+[nvidia-tesla.kernel-modules.nvidia.runtime-dependencies.nvidia]
+dependencies = ["nv-interface.o", "nv-kernel.o", "nvidia.mod.o"]
+
+[nvidia-tesla.kernel-modules.nvidia-modeset]
+dependencies = ["nvidia-modeset.mod.o"]
+
+[nvidia-tesla.kernel-modules.nvidia-modeset.runtime-dependencies.nvidia-modeset]
+dependencies = ["nv-modeset-interface.o", "nv-modeset-kernel.o"]
+
+[nvidia-tesla.kernel-modules.nvidia-uvm]
+dependencies = ["nvidia-uvm.o", "nvidia-uvm.mod.o"]
```

Comment:
I find it difficult to wrap my brain around this format, since although I know it translates into a series of `ld` invocations, it's not clear exactly which output files become input files.

I think it'd help to use the "real" file names, even though TOML makes that unlovely. I'd also recommend using the same paths as the dnf kmod packaging, so it's easier to compare with the `%post` scriptlet steps.

Here's an example of what I have in mind:
```
[nvidia-tesla.object-files."nvidia.o"]
link-objects = [
  "nvidia/nv-interface.o",
  "nvidia/nv-kernel.o"
]

[nvidia-tesla.object-files."nvidia-modeset.o"]
link-objects = [
  "nvidia-modeset/nv-modeset-interface.o",
  "nvidia-modeset/nv-modeset-kernel.o",
]

[nvidia-tesla.kernel-modules."nvidia.ko"]
link-objects = [
  "nvidia.o",
  "nvidia.mod.o",
]

[nvidia-tesla.kernel-modules."nvidia-modeset.ko"]
link-objects = [
  "nvidia-modeset.o",
  "nvidia-modeset.mod.o",
]
```

For each driver set, `driverdog` would first work through the set of object files (if any), then it would build the kernel modules.

That also addresses the (hypothetical) case where the same object needs to be built at runtime, and is used by two different kernel modules - the flatter structure allows us to specify the link steps and execute `ld` only once.